libtransmission/crypto-utils-ccrypto.cc

```diff
@@ -119,9 +119,18 @@ class Sha1Impl final : public tr_sha1
 
     void add(void const* data, size_t data_length) override
     {
+        auto const* sha_data = static_cast<uint8_t const*>(data);
+#if UINTPTR_MAX > UINT32_MAX
```

Comment:
I try to avoid ifdefs if possible; and even if we keep it, I'm not sure if `UINTPTR_MAX` is the right thing to check here?

Does this cause any warnings on macOS:

```c++
      void add(void const* data, size_t data_length) override
      {       
          auto const* sha_data = static_cast<uint8_t const*>(data);
          static auto constexpr Max = std::numeric_limits<CC_LONG>::max();
              
          while (data_length > 0)
          {   
              auto const n_bytes = static_cast<CC_LONG>(std::min(data_length, Max));
              CC_SHA1_Update(&handle_, sha_data, n_bytes);
              data_length -= n_bytes;
              sha_data += n_bytes;
          }
      }
```

Reply:
OK, code updated